pkg/commands/plan/plan_run.go

```diff
@@ -24,35 +24,195 @@ import (
 	"path"
 	"strconv"
 	"strings"
+	"time"
 
+	"github.com/abcxyz/guardian/pkg/flags"
+	"github.com/abcxyz/guardian/pkg/github"
 	"github.com/abcxyz/guardian/pkg/storage"
 	"github.com/abcxyz/guardian/pkg/terraform"
 	"github.com/abcxyz/guardian/pkg/util"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/sethvargo/go-githubactions"
 )
 
 const planCommentPrefix = "**`🔱 Guardian 🔱 PLAN`** -"
 
-// Result is the result of a plan operation.
-type Result struct {
+var _ cli.Command = (*PlanRunCommand)(nil)
+
+// RunResult is the result of a plan operation.
+type RunResult struct {
 	hasChanges     bool
 	commentDetails string
 }
 
-// Process handles the main logic for the Guardian plan process.
-func (c *PlanCommand) Process(ctx context.Context) error {
+type PlanRunCommand struct {
+	cli.BaseCommand
+
+	cfg *Config
+
+	planChildPath string
+	planFilename  string
+
+	flags.GitHubFlags
+	flags.RetryFlags
+
+	flagWorkingDirectory  string
+	flagBucketName        string
+	flagPullRequestNumber int
+	flagProtectLockfile   bool
+	flagLockTimeout       time.Duration
+
+	actions         *githubactions.Action
+	githubClient    github.GitHub
+	storageClient   storage.Storage
+	terraformClient terraform.Terraform
+
+	// testFlagSetOpts is only used for testing.
+	testFlagSetOpts []cli.Option
+}
+
+func (c *PlanRunCommand) Desc() string {
+	return `Run the Terraform plan for a directory.`
+}
+
+func (c *PlanRunCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Run the Terraform plan for a directory.
+`
+}
+
+func (c *PlanRunCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet(c.testFlagSetOpts...)
+
+	c.GitHubFlags.AddFlags(set)
+	c.RetryFlags.AddFlags(set)
+
+	f := set.NewSection("Command options")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "working-directory",
+		Target:  &c.flagWorkingDirectory,
+		Example: "terraform",
+		Usage:   "The working directory for Guardian to execute commands in.",
+	})
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "bucket-name",
+		Target:  &c.flagBucketName,
+		Example: "my-guardian-state-bucket",
+		Usage:   "The Google Cloud Storage bucket name to store Guardian plan files.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "protect-lockfile",
+		Target:  &c.flagProtectLockfile,
+		Default: true,
+		Example: "true",
```

Comment:
Same comment about booleans defaulting to false. This should be `-override-lockfile` and default to false.

Reply:
Went with --allow-lockfile-changes